EIPS/eip-3290.md

```diff
@@ -0,0 +1,172 @@
+
+7. DEFI Yielding
+Protect the deposited liquidity or the pool itself.
+
+## Specification
+TBS (Time Based Switch) feature integrates smart contracts and uses events/callbacks to inform the user about the TBS state.
+
+The user creates his TBS by calling the createSwitch function together with four parameters. This returns a signed transaction and puts it in a queue for due execution based on block height or a timestamp.
+
+```
+function createSwitch(uint _time, uint _amount, address_executor, address payable _benefitor)
+    doesntExist
+    checkAmount(_amount)
+    checkTime(_time)
+    public
+    Payable 
+{...}
+```
+
+TBS creation will produce a signed transaction that is put on hold until/if the trigger hits.
+All signed transactions are stored in the TBS queue object in the contract.
+
+Only the executor address is able to trigger a funds transfer when the defined timestamp has expired. The triggered event can be defined as any type of transaction or a specific transaction or blockchain information. Besides that, the creator will determine a percentage of balance dispersed per fallback address.
+
+``` 
+function tryExecuteSwitch(address account)
+    onlyValid(account)
+    onlyExecutorsOrOwner(account)
+    public
+    Payable
+{...}
+```
+
+Required event listener methods for tokens transfer are:
+```approveToken, withdrawToken```
+
+TBS will keep track/store all created switches to be later updated by the owners at will.
+The changeable variables would be:
+
+* Change function fallback addresses as owner
+* Change function condition variables as owner
+* Cancel TBS as an owner
+
+## Rationale
+This EIP standard is a ground layer with default functionality on top of which anyone could build a Dapp or integrate it in his product while allowing for safe and gas-efficient implementations. 
+
+TBS smart contracts should be implemented/imported as a library with only a couple of public functions. Beneficiary logic, notifications, and any other implementation (see motivation) can be abstracted from this standard and implemented on top of it. 
+
+Here are some of the different approaches may include this standard:
+
+* Custom wallet integration for TBS setup
+* (De)centralized notifications/reminders before an action occur.
+* Custom fallback consensus and validation logic
+* Multi TBS setup
+* NFT transfers 
+
+We also considered a solution that triggers the tokens transfer. Still, after long consideration, we decided that it should be implemented on the application layer because we wanted to leave a more extensive range of integration with this standard. 
+
+## Backwards Compatibility
+This standard is compatible with any fungible and non-fungible token, such as ERC-20 or ERC-721.
+
+## Reference Implementation
+https://github.com/Omodaka9375/Time-Based-Switch-on-Ethereum/blob/main/contracts/TimeBasedSwitch.sol
```

Comment:
This code should be inlined or included in `../assets/3290/TimeBasedSWitch.sol` and then linked from this document via relative link.